File: BlogApp_Crud_Rest_Api_Express_Flutter/frontend/lib/domain/entities/post.dart
// File: lib/domain/entities/post.dart
class Post {
  final String id;
  final String userId;
  final String? text;
  final String? image;
  final String? address;
  final String createdAt;
  final String updatedAt;
  final String authorId;
  final String authorName;

  Post({
    required this.id,
    required this.userId,
    this.text,
    this.image,
    this.address,
    required this.createdAt,
    required this.updatedAt,
    required this.authorId,
    required this.authorName,
  });
}

File: BlogApp_Crud_Rest_Api_Express_Flutter/backend/domain/entities/Post.js
// domain/entities/Post.js

class Post {
    constructor({ id, title, body, author, authorId, createdAt, updatedAt }) {
      this.id = id;
      this.title = title;
      this.body = body;
      this.author = author;
      this.authorId = authorId;
      this.createdAt = createdAt;
      this.updatedAt = updatedAt;
    }
  }
  
  module.exports = Post;
  